src/coreclr/gc/gc.cpp

```diff
@@ -39572,14 +39570,55 @@ void gc_heap::decommit_ephemeral_segment_pages()
     }
 
 #if defined(MULTIPLE_HEAPS) && defined(USE_REGIONS)
-    // for regions, this is done at the regions level
-    return;
+    for (int gen_number = soh_gen0; gen_number <= soh_gen1; gen_number++)
```

Comment:
if we are not even doing a gen1 GC, would we need to goto `soh_gen1` here? 

and I presume you are going to gen2 because gen2 budget is usually too large to make this attempt worthwhile?

if we are doing a gen1 GC we would have called `decommit_heap_segment_pages` and decommitted (almost) everything after `heap_segment_allocated` for gen1 regions -
```cplusplus
            if (gen_num != 0)
            {
                dprintf (REGIONS_LOG, ("  gen%d decommit end of region %Ix(%Ix)",
                    gen_num, current_region, heap_segment_mem (current_region)));
                decommit_heap_segment_pages (current_region, 0);
            }
```
if `dt_high_memory_load_p()` is true. so you could avoid doing the gen1 region in that case. however, I'm also fine if you want to avoid complicating this code path here.

Reply:
There are two reasons for considering the situation in `soh_gen1` even if we are not doing a gen 1 GC:
- The free list situation in gen 1 could have changed.
- We are smoothing the ramp down, and gen 1 GCs may happen too infrequently. The alternative would be to not do the smoothing, but I think that would likely perform worse in cases where the work load changes behavior.

We need to consider what to do with gen 2 (and UOH as well), agreed. This should be a separate PR, I think.

Reply:
actually I think for now we should keep the behavior for `decommit_heap_segment_pages` for now, for gen2/UOH - there's no reason not to because that's only done during a full blocking GC. and it's the same behavior for segments. we can make a separate PR to improve this behavior.